Structure_Engine/Query/ReinforcingBars.cs

```diff
@@ -0,0 +1,107 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.SectionProperties.Reinforcement;
+using BH.oM.Structure.Elements;
+using BH.oM.Physical.Reinforcement;
+using BH.oM.Physical.Materials;
+using BH.oM.Spatial.Layouts;
+using BH.oM.Geometry;
+using BH.Engine.Spatial;
+using BH.Engine.Geometry;
+using BH.oM.Base;
+
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Extract all ReinforcingBars from the structural Bar. Only extract reinforcement for bars owning a ConcreteSection, for other sectiontypes, an empty list will be returned.")]
+        [Input("bar", "The structural Bar to extract Reinforcement from.")]
+        [Output("rebars", "All ReinforcingBar on the provided Bar.")]
+        public static List<IReinforcingBar> ReinforcingBars(this Bar bar)
```

Comment:
I would highlight in some manner that we're going from Structure objects to Physical objects. 
The difference between `IBarReinforcement` and `IReinforcingBar` is not big enough on its own